src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -0,0 +1,1564 @@
+        {
+            return new LexicalSortKey(Location, this.DeclaringCompilation);
+        }
+
+        public override ImmutableArray<Location> Locations
+        {
+            get
+            {
+                return ImmutableArray.Create(Location);
+            }
+        }
+
+        public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences
+        {
+            get
+            {
+                return ImmutableArray.Create(_syntaxRef);
+            }
+        }
+
+        public override bool IsAbstract
+        {
+            get { return (_modifiers & DeclarationModifiers.Abstract) != 0; }
+        }
+
+        public override bool IsExtern
+        {
+            get { return (_modifiers & DeclarationModifiers.Extern) != 0; }
+        }
+
+        public override bool IsStatic
+        {
+            get { return (_modifiers & DeclarationModifiers.Static) != 0; }
+        }
+
+        internal bool IsFixed
+        {
+            get { return false; }
+        }
+
+        /// <remarks>
+        /// Even though it is declared with an IndexerDeclarationSyntax, an explicit
+        /// interface implementation is not an indexer because it will not cause the
+        /// containing type to be emitted with a DefaultMemberAttribute (and even if
+        /// there is another indexer, the name of the explicit implementation won't
+        /// match).  This is important for round-tripping.
+        /// </remarks>
+        public override bool IsIndexer
+        {
+            get { return (_modifiers & DeclarationModifiers.Indexer) != 0; }
+        }
+
+        public override bool IsOverride
+        {
+            get { return (_modifiers & DeclarationModifiers.Override) != 0; }
+        }
+
+        public override bool IsSealed
+        {
+            get { return (_modifiers & DeclarationModifiers.Sealed) != 0; }
+        }
+
+        public override bool IsVirtual
+        {
+            get { return (_modifiers & DeclarationModifiers.Virtual) != 0; }
+        }
+
+        internal bool IsNew
+        {
+            get { return (_modifiers & DeclarationModifiers.New) != 0; }
+        }
+
+        internal bool HasReadOnlyModifier => (_modifiers & DeclarationModifiers.ReadOnly) != 0;
+
+        public override MethodSymbol GetMethod
+        {
+            get { return _getMethod; }
+        }
+
+        public override MethodSymbol SetMethod
+        {
+            get { return _setMethod; }
+        }
+
+        internal override Microsoft.Cci.CallingConvention CallingConvention
+        {
+            get { return (IsStatic ? 0 : Microsoft.Cci.CallingConvention.HasThis); }
+        }
+
+        public override ImmutableArray<ParameterSymbol> Parameters
+        {
+            get
+            {
+                if (_lazyParameters.IsDefault)
+                {
+                    var diagnostics = DiagnosticBag.GetInstance();
+                    var binder = this.CreateBinderForTypeAndParameters();
```

Comment:
>var binder = this.CreateBinderForTypeAndParameters(); [](start = 20, length = 53)

I think the logic to compute parameters and getting the binder for that should be moved into an abstract method.